### PR TITLE
refactor: scm webhook trigger

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Refactor scm webhook trigger, try to decouple scm webhook with integration

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/meta/labels.go [hunk 1]
```diff
@@ -41,6 +41,12 @@ const (
 	// LabelIntegrationSchedulableCluster is the label key used to indicate the cluster is schedulable for workflowruns in this tenant
 	LabelIntegrationSchedulableCluster = "integration.cyclone.dev/schedulable-cluster"
 
+	// LabelIntegrationOfWft is the label key used to indicate integration name of wft
+	LabelIntegrationOfWft = "integration.workflowtrigger.cyclone.dev/name"
```
Comment (cd1989) [inline pkg/meta/labels.go]: How about, `LabelWftEventSource = workflowtrigger.cyclone.dev/event-source`

File: pkg/meta/labels.go [hunk 2]
```diff
@@ -41,6 +41,12 @@ const (
...
+
+	// LabelRepoOfWft is the label key used to indicate scm repo name of wft
+	LabelRepoOfWft = "repo.workflowtrigger.cyclone.dev/name"
```
Comment (cd1989) [inline pkg/meta/labels.go]: `LabelWfrEventRepo` = "workflowtrigger.cyclone.dev/event-repo"

File: pkg/server/apis/v1alpha1/descriptors/webhook.go [hunk 3]
```diff
@@ -26,9 +26,13 @@ var webhook = []definition.Descriptor{
 						Description: "tenant",
 					},
 					{
-						Source:      definition.Path,
-						Name:        "integration",
-						Description: "Integration name",
+						Source:      definition.Query,
+						Name:        "eventType",
+						Description: "the webhook eventType, support SCM for now",
```
Comment (cd1989) [inline pkg/server/apis/v1alpha1/descriptors/webhook.go]: I'd prefer `sourceType`, for a given event source, there may be different event types.

File: pkg/server/biz/hook/hook.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	"github.com/caicloud/cyclone/pkg/apis/cyclone/v1alpha1"
+	"github.com/caicloud/cyclone/pkg/meta"
+	api "github.com/caicloud/cyclone/pkg/server/apis/v1alpha1"
+	in "github.com/caicloud/cyclone/pkg/server/biz/integration"
+	"github.com/caicloud/cyclone/pkg/server/biz/scm"
+	"github.com/caicloud/cyclone/pkg/server/common"
+	"github.com/caicloud/cyclone/pkg/server/config"
+	"github.com/caicloud/cyclone/pkg/server/handler"
+	"github.com/caicloud/cyclone/pkg/util/cerr"
+	"github.com/caicloud/nirvana/log"
```
Comment (cd1989) [inline pkg/server/biz/hook/hook.go]: Not well formated

File: pkg/server/biz/hook/hook.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+	if o == nil {
+		o = &Operator{
+			mutex: sync.Mutex{},
+		}
+	}
+
+	return o
+}
+
+// ListWfts list all relative workflow triggers
```
Comment (cd1989) [inline pkg/server/biz/hook/hook.go]: `related` ?

File: pkg/server/biz/hook/hook.go [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+	wfts, err := handler.K8sClient.CycloneV1alpha1().WorkflowTriggers(common.TenantNamespace(tenant)).List(listOption)
+	if err != nil {
+		return nil, err
+	}
+
+	return wfts, nil
+}
+
+// RegisterSCMWebhook registers SCM webhook if if has not been registered.
+func (*Operator) RegisterSCMWebhook(tenant, wftName, secretName, repo string) error {
```
Comment (cd1989) [inline pkg/server/biz/hook/hook.go]: I'd prefer to describe a interface like `Manager` (under package `hook`). With methods like `Register(...)`, `UnRegister(...)`.

Then make implementation like `SCMManager`...

**Comment by zhujian7** [comment]
@cd1989 PTAL

**Pull request merged**